File: leetcode.com/Python/1512_NumberofGoodPairs.py
# 1512. Number of Good Pairs
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count, res = [0] * (max(nums) + 1), 0
        for num in nums:
            count[num] += 1
        for n in count:
            res += int(n * (n-1)/2)
        return res

File: leetcode.com/Java/1512_NumberofGoodPairs.java
// 1512. Number of Good Pairs
class Solution {

    public int numIdenticalPairs(int[] nums) {
        int N = 0, res = 0;
        for (int x : nums) N = x > N ? x : N;
        int[] count = new int[N + 1];
        for (int num : nums) count[num]++;
        for (int n : count) res += (n * (int) 1.0 * (n - 1)) / 2;
        return res;
    }
}
